### ISSUE TITLE
"Mock" is prefixed even when using `mock!`

### ISSUE THREAD

**Issue by garrettmaring** [asomers/mockall#331]
in the [automock! documentation](https://docs.rs/mockall/0.10.2/mockall/attr.automock.html) it is suggested to use `mock!` if "the autogenerated “MockFoo” name isn’t acceptable, and you want to choose your own name for the mock structure."

however, when I use `mock! { pub MyStruct {} }` i received an undeclared variable when trying to read MyStruct but it works when I change my import to use MockMyStruct.

how can I determine the name of the mocked structure using `mock!`?

**Comment by asomers** [comment]
You're right; you can't.  `mock!` is designed to allow as closely as possible copy/pasting the original struct's definition.  That's why it prefixes the "Mock".  Do you have a good suggestion for a syntax for an enhancement that would disable the automatic prefix?

**Comment by garrettmaring** [comment]
hm, is `mock!(prefix = "")` the simplest possibility?

i'm also reminded of serde's renaming `#[serde(rename = "aaa")]` which could lead to something similar that might go inside `mock!` above the struct definition.
